serverless.yml

```
@@ -18,96 +18,86 @@ service: remote-patient-monitoring
+    DebugMode: ${file(config/${self:provider.stage}.json):DebugMode}
 
 # you can add packaging information here
-#package:
-#  include:
-#    - include-me.js
-#    - include-me-dir/**
-#  exclude:
-#    - exclude-me.js
-#    - exclude-me-dir/**
+package:
+  #  include:
+  #    - include-me.js
+  #    - include-me-dir/**
+  exclude:
+    - docs/**
+    - templates/**
+    - .eslintrc.json
+    - LICENSE
+    - README.md
+    - yarn.lock
+    - package.json
 
-functions:
-  hello:
-    handler: handler.hello
-#    The following are a few example events you can configure
-#    NOTE: Please make sure to change your handler code to work with those events
-#    Check the event documentation for details
-#    events:
-#      - http:
-#          path: users/create
-#          method: get
-#      - websocket: $connect
-#      - s3: ${env:BUCKET}
-#      - schedule: rate(10 minutes)
-#      - sns: greeter-topic
-#      - stream: arn:aws:dynamodb:region:XXXXXX:table/foo/stream/1970-01-01T00:00:00.000
-#      - alexaSkill: amzn1.ask.skill.xx-xx-xx-xx
-#      - alexaSmartHome: amzn1.ask.skill.xx-xx-xx-xx
-#      - iot:
-#          sql: "SELECT * FROM 'some_topic'"
-#      - cloudwatchEvent:
-#          event:
-#            source:
-#              - "aws.ec2"
-#            detail-type:
-#              - "EC2 Instance State-change Notification"
-#            detail:
-#              state:
-#                - pending
-#      - cloudwatchLog: '/aws/lambda/hello'
-#      - cognitoUserPool:
-#          pool: MyUserPool
-#          trigger: PreSignUp
-#      - alb:
-#          listenerArn: arn:aws:elasticloadbalancing:us-east-1:XXXXXX:listener/app/my-load-balancer/50dc6c495c0c9188/
-#          priority: 1
-#          conditions:
-#            host: example.com
-#            path: /hello
-
-#    Define function environment variables here
-#    environment:
-#      variable2: value2
+functions: ${file(./templates/functions.yml)}
 
 # you can add CloudFormation resource templates here
-#resources:
-#  Resources:
-#    NewResource:
-#      Type: AWS::S3::Bucket
-#      Properties:
-#        BucketName: my-new-bucket
-#  Outputs:
-#     NewOutput:
-#       Description: "Description for the output"
-#       Value: "Some output value"
+resources:
+  # Please comment out below two lines for the first deploy. After the deployment, enable it and deploy again.
+  - ${file(./templates/api-gateway.yml)}
+  - ${file(./templates/swagger-ui.yml)}
+  # Swagger UI 関連の設計は https://betterdev.blog/serverless-swagger-ui-for-api-gateway/ を参考にした。
+
+plugins:
+  - serverless-pseudo-parameters
+  - serverless-s3-deploy
+  - serverless-scriptable-plugin
+  - serverless-cloudformation-sub-variables
+  - serverless-s3-remover
+custom:
+  project: remote-patient-monitoring
+  baseName: ${self:custom.project}-${self:provider.stage}-${self:service}
+  scriptHooks:
+    after:aws:deploy:deploy:updateStack:
+      - generateConfigFile.js
+      - webpack --mode production --progress
```

```suggestion
      - ./node_modules/.bin/webpack --mode production --progress
```
globalにwebpackがインストールされていない場合、こうしないとエラーになるようでした